with-msw: Module not found: Package path ./lib is not exported from package

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 21.6.0: Mon Aug 22 20:17:10 PDT 2022; root:xnu-8020.140.49~2/RELEASE_X86_64
    Binaries:
      Node: 16.17.0
      npm: 8.19.1
      Yarn: 1.22.19
      pnpm: 7.12.1
    Relevant packages:
      next: 12.3.1
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0

### Which example does this report relate to?

with-msw

### What browser are you using? (if relevant)

Version 105.0.5195.125

### How are you deploying your application? (if relevant)

_No response_

### Describe the Bug

- When I start the app I am getting this error:

```sh
error - ./node_modules/msw/lib/index.js:455:0
Module not found: Package path ./lib is not exported from package /Users/elbertcastaneda/workspaces/softwareLab/js/nextjs-with-msw-app/node_modules/headers-polyfill (see exports field in /Users/elbertcastaneda/workspaces/softwareLab/js/nextjs-with-msw-app/node_modules/headers-polyfill/package.json)

Import trace for requested module:
./mocks/browser.ts
./mocks/index.ts
./pages/_app.tsx

https://nextjs.org/docs/messages/module-not-found
```

### Expected Behavior

The app runs correctly

### To Reproduce

- yarn create next-app --example with-msw with-msw-app
- cd with-msw-app
- yarn run dev

--- a/examples/with-msw/package.json
+++ b/examples/with-msw/package.json
@@ -6,7 +6,7 @@
     "start": "next start"
   },
   "dependencies": {
-    "msw": "^0.44.2",
+    "msw": "0.47.3",
     "next": "latest",
     "react": "^18.2.0",
     "react-dom": "^18.2.0"

